test/xdg_desktop_portal_test.dart

```diff
@@ -2244,6 +2296,32 @@ void main() {
         equals(['http://localhost:1234']));
   });
 
+  test('secret', () async {
+    var server = DBusServer();
+    var clientAddress =
+        await server.listenAddress(DBusAddress.unix(dir: Directory.systemTemp));
+    addTearDown(() async {
+      await server.close();
+    });
+
+    var portalServer = MockPortalServer(clientAddress);
+    await portalServer.start();
+    addTearDown(() async {
+      await portalServer.close();
+    });
+
+    var client = XdgDesktopPortalClient(bus: DBusClient(clientAddress));
+    addTearDown(() async {
+      await client.close();
+    });
+
+    expect(await client.secret.getVersion(), equals(1));
+
+    final secret = await client.secret.retrieveSecret();
+    expect(portalServer.secret, equals([MockSecret({})]));
```

Comment:
The test should set the secret value in the `MockPortalServer` constructor and check that value is returned here. i.e. similar to how a `networkMetered` is set.